src/devices/UFireIse/UFirePh .cs

```diff
@@ -0,0 +1,172 @@
+                {
+                    temperatureMultiplier *= -1;
+                }
+
+                if ((Ph <= 6.0) && (temp.Value.DegreesCelsius <= 15))
+                {
+                    temperatureMultiplier *= -1;
+                }
+
+                Ph += Convert.ToSingle(temperatureMultiplier);
+            }
+
+            if (Ph <= 0.0 || Ph >= 14.0)
+            {
+                pH = float.NaN;
+                Ph = pH;
+
+                return false;
+            }
+
+            if (float.IsNaN(Ph) || float.IsInfinity(Ph))
+            {
+                pH = float.NaN;
+                Ph = pH;
+
+                return false;
+            }
+
+            Ph = pH;
+
+            return true;
+        }
+
+        /// <summary>
+        /// Calibrates the probe using a single point using a pH value.
+        /// </summary>
+        /// <param name="solutionpH">pH value</param>
+        public void CalibrateSingle(float solutionpH)
+        {
+            CalibrateSingle(PHtomV(solutionpH));
+        }
+
+        /// <summary>
+        /// Calibrates the dual-point values for the high reading and saves them in the devices's EEPROM.
+        /// </summary>
+        /// <param name="solutionpH">The pH of the calibration solution</param>
+        public void CalibrateProbeHigh(float solutionpH)
+        {
+            CalibrateProbeHigh(PHtomV(solutionpH));
+        }
+
+        /// <summary>
+        /// Returns the dual-point calibration high-reference value.
+        /// </summary>
+        /// <returns></returns>
+        public new float GetCalibrateHighReference()
+        {
+            return MVtopH(GetCalibrateHighReference());
+        }
+
+        /// <summary>
+        /// Returns the dual-point calibration high-reading value.
+        /// </summary>
+        /// <returns></returns>
+        public new float GetCalibrateHighReading()
+        {
+            return MVtopH(GetCalibrateHighReading());
+        }
+
+        /// <summary>
+        /// Calibrates the dual-point values for the low reading and saves them in the devices's EEPROM.
+        /// </summary>
+        /// <param name="solutionpH"> the pH of the calibration solution</param>
+        public void CalibrateProbeLow(float solutionpH)
+        {
+            CalibrateProbeLow(PHtomV(solutionpH));
+        }
+
+        /// <summary>
+        /// Returns the dual-point calibration low-reference value.
+        /// </summary>
+        /// <returns></returns>
+        public new float GetCalibrateLowReference()
+        {
+            return MVtopH(GetCalibrateLowReference());
+        }
+
+        /// <summary>
+        /// Returns the dual-point calibration low-reading value.
+        /// </summary>
+        /// <returns></returns>
+        public new float GetCalibrateLowReading()
+        {
+            return MVtopH(GetCalibrateLowReading());
+        }
+
+        private float PHtomV(float pH)
```

Comment:
PhToMillivolts - this name took me couple of seconds to figure out what it means